homeassistant/components/switchbot_cloud/vacuum.py

```diff
@@ -0,0 +1,113 @@
+"""Support for SwitchBot vacuum."""
+
+from typing import Any
+
+from switchbot_api import Device, Remote, SwitchBotAPI, VacuumCommands
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+from .entity import SwitchBotCloudEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        _async_make_entity(data.api, device, coordinator)
+        for device, coordinator in data.devices.vacuums
+    )
+
+
+# vacuum state / workingStatus
+# "StandBy": "Stand by",
+# "Clearing": "Clearing",
+# "Paused": "Paused",
+# "GotoChargeBase": "Go to charge base",
+# "Charging": "Charging",
+# "ChargeDone": "Charge done",
+# "Dormant": "Dormant",
+# "InTrouble": "In trouble",
+# "InRemoteControl": "In remote control",
+# "InDustCollecting": "In dust collecting"
+
+# fan speed
+# "0": "Quiet",
+# "1": "Standard",
+# "2": "Strong",
+# "3": "MAX"
+
+
+# GET /v1.1/devices/{deviceId}/status
+# https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#robot-vacuum-cleaner-s1-plus-1
+class SwitchBotCloudVacuum(SwitchBotCloudEntity, StateVacuumEntity):
+    """Representation of a SwitchBot vacuum."""
+
+    _attr_supported_features: VacuumEntityFeature = (
+        VacuumEntityFeature.BATTERY
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.START
+        | VacuumEntityFeature.STATE
+    )
+
+    _attr_name = None
+    _attr_fan_speed_list: list[str] = ["1", "2", "3", "4"]
```

Comment:
Can we make these snake_cased names? Like `quiet`? This way we can translate them properly and the values still make sense when templating

Reply:
Yes, it's possible. I'll make the change

Reply:
Ok, I made the change